File: Student Attendance Record I.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        absent = 0
        late = 0
        for c in s:
            if c == "L":
                late += 1
                if late == 3:
                    return False
            else:
                late = 0
                absent += int(c == 'A')
        return absent <= 1

File: Student Attendance Record I.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int late = 0, absent = 0;
        for (char& c: s) {
            if (c == 'L') {
                late++;
                if (late==3) return false;
            } else {
                late = 0;
                absent += c == 'A';
            }
        }
        return absent <= 1;
    }
};
